src/components/TextField/TextField.tsx

```diff
@@ -38,50 +74,89 @@ export class TextField extends React.Component<TextFieldProps> {
     this.TextFieldRef.current.blur();
   };
 
-  getSuffix = () => {
+  _getSuffix = () => {
     const {
+      error,
       errorMessage,
-      success = false,
-      successIcon = false,
-      error = false,
+      success,
+      successIcon,
       suffix,
+      withClearButton,
+      clearButtonAriaLabel,
+      clearButtonAriaLabelledby,
+      onClear,
+      value,
+      disabled,
     } = this.props;
-    let suffixToShow = suffix;
-
-    if (errorMessage && error) {
-      suffixToShow = (
-        <Tooltip
-          appendTo="scrollParent"
-          placement="top-end"
-          skin={TooltipSkin.Error}
-          content={errorMessage}
-          moveBy={{ x: 5, y: 0 }}
-        >
-          <ErrorIcon />
-        </Tooltip>
-      );
-    }
-
-    if (successIcon && success) {
-      suffixToShow = <SuccessIcon data-hook="successIcon" />;
-    }
-    return suffixToShow;
+
+    const shouldShowCustomSuffix = !!suffix;
+    const shouldShowErrorIcon = error && errorMessage;
+    const shouldShowSuccessIcon = successIcon && success;
+    const shouldShowClearButton = withClearButton && value && !disabled;
+
+    const hasSuffix =
+      shouldShowCustomSuffix ||
+      shouldShowErrorIcon ||
+      shouldShowSuccessIcon ||
+      shouldShowClearButton;
+
+    const shouldAddLeftPaddingToCustomSuffix =
+      !shouldShowErrorIcon && !shouldShowSuccessIcon && shouldShowClearButton;
```

Comment:
I think instead of using these states to understand if there needs to be padding or not, I would give all the icons a className like `suffixIcon` and go with a pure css solution like this:
```
.suffixWrapper .suffixIcon {
  padding-right: number;
}

.suffixWrapper .suffixIcon::last-child {
  padding-right: none;
}
```
Haven't tested this of course, so I'm not 100% sure this will work

Reply:
btw, how does this work with `rtl`?

Reply:
I refactored it and fix the icons spaces